src/sgm_stereo_node.cpp

```diff
@@ -301,6 +290,104 @@ void StereoSGMNode::computePointCloudFromDisparity( const sensor_msgs::ImageCons
   }
 }
 
+void StereoSGMNode::extractTextureLessRegion(const cv::Mat& edge_img, cv::Mat& textureless_mask)
+{
+  for (int x = 0; x < edge_img.cols; x++)
+  {
+    for (int y = 0; y < edge_img.rows; y++)
+    {
+      if (sumNeighborEdgeGradients(edge_img, x, y) >= textureless_thresh_)
+      {
+        textureless_mask.at<uchar>(y, x) = 255;
+      }
+    }
+  }
+
+  if (display_debugging_images_)
+  {
+    if (rotate_images_)
+    {
+      cv::Mat textureless_mask_rot = textureless_mask.clone();
+      cv::flip(textureless_mask_rot, textureless_mask_rot, -1);
+      cv::imshow("textureless_mask", textureless_mask_rot);
+
+    }
+    else
+    {
+      cv::imshow("textureless_mask", textureless_mask);
+    }
+    cv::waitKey(10);
+  }
+}
+
+int StereoSGMNode::sumNeighborEdgeGradients(const cv::Mat& edge_img, int x, int y)
+{
+  int x_inds[] = {-2, -1, 0, 1, 2,        -2, -1, 0, 1, 2,       -2, -1, 0, 1, 2,     -2, -1, 0, 1, 2,      -2, -1, 0, 1, 2};
+  int y_inds[] = { 2,  2, 2, 2, 2,         1,  1, 1, 1, 1,        0,  0, 0, 0, 0,     -1, -1, -1, -1, -1,   -2, -2, -2, -2, -2};
+
+  // TODO: make neighborhood size configurable
+  int n = 25;
+
+  int sum_region_gray = 0;
+
+  for (int i = 0; i < n; i++)
+  {
+    int nx = x + x_inds[i];
+    int ny = y + y_inds[i];
+
+    // Check if pixels out of bounds
+    if (nx < 0 || nx >= edge_img.cols || ny < 0 || ny >= edge_img.rows)
+    {
+      continue;
+    }
+
+    uchar gray = edge_img.at<uchar>(ny, nx);
+
+    sum_region_gray += (int)gray;
+  }
+
+  return sum_region_gray;
+}
+
+
+void StereoSGMNode::extractSobelVerticalEdges(const cv::Mat& gray_img, std::string title, cv::Mat& grad)
```

Comment:
Is it faster to do this step with OpenCV rather than the original SGM code?

Reply:
But why would I want to extract the SGM code here?

Reply:
I thought you re-implemeted this using OpenCV functions to replace it in SGM. Then I realized you use it for the textureless filter.

Not for this PR, but maybe you can consolidate this so you use the same for both (if they're exactly the same), and use the fastest one, and don't do it twice.

Maybe just add a TODO comment.